### PR TITLE
Oheger bosch/scanner/clearly defined helper functionality

### PR DESCRIPTION
**Pull request by oheger-bosch**
This PR adds some minor functionality, which is going to be needed by the implementation of the ClearlyDefinedStorage. It includes the following changes:

* A function isCompatibleVersion() is extracted out of the ScannerDetails class, which is going to be used to check for the version compatibility of scan results, independent on other properties.
* Support for another query endpoint has been added to the ClearlyDefinedService class. This endpoint is going to be used to find all the results available related to a specific package identifier.

### CODE REVIEW

File: clearly-defined/src/main/kotlin/ClearlyDefinedService.kt [hunk 1]
```diff
@@ -409,6 +410,19 @@ interface ClearlyDefinedService {
     @POST("definitions")
     fun getDefinitions(@Body coordinates: Collection<String>): Call<Map<String, Defined>>
 
+    /**
+     * Search for existing definitions based on the [pattern] string provided. The pattern string should contain
+     * the parts of the coordinates (typically namespace, name, and version) relevant for the search. Result is a
+     * list with the ClearlyDefined URIs to all the definitions that are matched by the pattern.
+     *
+     * Note that the _/definitions_ GET endpoint behaves differently depending on the parameters passed in. If only
```
Comment (sschuberth) [inline clearly-defined/src/main/kotlin/ClearlyDefinedService.kt:418]: Formatting nit: Maybe better `"definitions" endpoint`? Also: What "parameters" (plural)? The function only takes *one* "pattern" parameter. If you're referring to the (potentially multiple) coordinate components mentioned above, then maybe better say "depending on the coordinates passed in".

File: clearly-defined/src/main/kotlin/ClearlyDefinedService.kt [hunk 2]
```diff
@@ -409,6 +410,19 @@ interface ClearlyDefinedService {
...
+     * a pattern string is provided, result is a list of URIs - this use case is covered by this function. It is
```
Comment (sschuberth) [inline clearly-defined/src/main/kotlin/ClearlyDefinedService.kt:419]: "..., *the* result is ..."

File: clearly-defined/src/main/kotlin/ClearlyDefinedService.kt [hunk 3]
```diff
@@ -409,6 +410,19 @@ interface ClearlyDefinedService {
...
+     * also possible to provide other parameters like type, provider, namespace, name, etc. Then the endpoint
+     * returns partly filled definition documents.
```
Comment (sschuberth) [inline clearly-defined/src/main/kotlin/ClearlyDefinedService.kt:421]: IMO it's confusing that you seem to describe parameters that the `definitions` entry point theoretically supports, but this function does not. You should only describe what *this* function supports / does.

Also, I'm missing a link to the upstream API docs like we have for the other functions.

**Review by sschuberth — CHANGES_REQUESTED** [review]

File: clearly-defined/src/main/kotlin/ClearlyDefinedService.kt (hunk 1 above)
Comment (oheger-bosch) [reply clearly-defined/src/main/kotlin/ClearlyDefinedService.kt]: Reworded and dropped the part which is not relevant to this function.

File: clearly-defined/src/main/kotlin/ClearlyDefinedService.kt (hunk 2 above)
Comment (oheger-bosch) [reply clearly-defined/src/main/kotlin/ClearlyDefinedService.kt]: Reworded and dropped the part which is not relevant to this function.

File: clearly-defined/src/main/kotlin/ClearlyDefinedService.kt (hunk 3 above)
Comment (oheger-bosch) [reply clearly-defined/src/main/kotlin/ClearlyDefinedService.kt]: Reworded and added the link. The comment now only describes the relevant part for this function.

I have to admit, I was also confused by the API of ClearlyDefined in this case. Actually, there are two different functionalities behind the definitions GET endpoint:
* You can pass in a pattern string and get back a list of matching URIs or
* you can pass in a number of parameters defining package coordinates and get back partly filled definition objects.

As we only use the first functionality, I stripped down the function comment to this use case as requested.

**Review by sschuberth — APPROVED** [review]

**Pull request merged**